momepy/preprocessing.py

```diff
@@ -740,3 +747,350 @@ def _get_extrapolated_line(coords, tolerance, point=False):
+            edges.geometry, op="covered_by"
+        )
+    idx_drop = edges.index.take(edges_idx)
+    touching_idx = touching.index
+    ls = list(set(touching_idx) - set(idx_drop))
+
+    incoming = touching.loc[ls]
+
+    # figuring out which ends of incoming edges need to be connected to the center_pt
+    incoming["first_pt"] = incoming.geometry.apply(lambda x: Point(x.coords[0]))
+    incoming["dist_first_pt"] = incoming.center_pt.distance(incoming.first_pt)
+    incoming["last_pt"] = incoming.geometry.apply(lambda x: Point(x.coords[-1]))
+    incoming["dist_last_pt"] = incoming.center_pt.distance(incoming.last_pt)
+    lines = []
+    for i, row in incoming.iterrows():
+        if row.dist_first_pt < row.dist_last_pt:
+            lines.append(LineString([row.first_pt, row.center_pt]))
+        else:
+            lines.append(LineString([row.last_pt, row.center_pt]))
+    incoming["line"] = gpd.GeoSeries(lines, index=incoming.index, crs=edges.crs)
+
+    # checking if there are more than one incoming lines arriving to the same point
+    # which would create several new lines
+    incoming["line_wkt"] = incoming.line.to_wkt()
+    grouped_lines = incoming.groupby(["line_wkt"])["line_wkt"]
+    count_s = grouped_lines.count()
+
+    # separating the incoming roads that come on their own to those that come in groups
+    filter_count_one = pd.DataFrame(count_s[count_s == 1])
+    filter_count_many = pd.DataFrame(count_s[count_s > 1])
+    incoming_ones = pd.merge(
+        incoming, filter_count_one, left_on="line_wkt", right_index=True, how="inner"
+    )
+    incoming_many = pd.merge(
+        incoming, filter_count_many, left_on="line_wkt", right_index=True, how="inner"
+    )
+    incoming_many_reduced = _coins_filtering_many_incoming(
+        incoming_many, angle_threshold=angle_threshold
+    )
+
+    incoming_all = gpd.GeoDataFrame(
+        pd.concat([incoming_ones, incoming_many_reduced]), crs=edges.crs
+    )
+
+    return incoming_all, idx_drop
+
+
+def _ext_lines_to_center(edges, incoming_all, idx_out):
+    """
+    Extends the LineStrings geometries to the centerpoint defined by
+    _rabs_center_points. Also deletes the lines that originally defined the roundabout.
+    Creates a new column labled with the 'rab' number.
+
+    Returns
+    -------
+    GeoDataFrame
+        GeoDataFrame with updated geometry
+    """
+
+    incoming_all["geometry"] = incoming_all.apply(
+        lambda row: linemerge([row.geometry, row.line]), axis=1
+    )
+
+    new_edges = edges.drop(idx_out, axis=0)
+
+    # creating a unique label for returned gdf
+    incoming_label = "rab_" + incoming_all.index_right.astype(str)
+    incoming_label = incoming_label[~incoming_label.index.duplicated(keep="first")]
+
+    # maintaining the same gdf shape as the original
+    incoming_all = incoming_all[edges.columns]
+    new_edges = pd.concat([new_edges, incoming_all])
+
+    # adding a new column to match
+    new_edges["simpl_edge"] = incoming_label
```

Comment:
`"simpl_edge"` is not a very intuitive name. Same as the value that comes with it (i.e. `rab_12`). It should be clear what it is and what it refers to. What is `12` in `rab_12`? Why do I get 3 groups if I use your testing data that are `rab_6`, `rab_8` and `rab_12`? Where does this come from in relation to my original data? Why do I get it back?

Reply:
Let me explain further to come up with better namings.
`simple_edge` --> short for 'simplified_edge' : This is a column that's aimed at identifying the edges that have been altered during this process. In my head, the upcoming methods() will further populate this same column. 

`rab_` --> short for 'roundabout'. The number comes originally from the `id` of the roundabout polygon which is used to identify the overall group (including adjacent polygons when appropriate).

Reply:
Okay. I think that if we're adding this as a column to the data that didn't have it previously, it needs to be useful for the user. I would call the column `simplification_group` and try to return numerical ids, not strings with rab_. Why are some values missing? I would expect I'd get 1, 2, 3, not 6, 8, 12.

Reply:
just made said changes ...